upstream/uphttp/upstream.go

```diff
@@ -20,20 +20,35 @@ import (
 )
 
 type Upstream struct {
-	httpOut httpout.Service
-	devMode bool
+	httpOut    httpout.Service
+	appRootURL func(_ apps.App, path string) (string, error)
```

Comment:
What is the point here? Calling `AppRootURL` on every `Roundtrip` easier?

Reply:
This is in prep for OpenFaas that wraps the http upstream, but provides the RootURL of its own, dynamically